src/ml_board/frontend/Dashboard/src/components/graphs/Graph.tsx

```diff
@@ -43,8 +46,58 @@ export default function Graph({ chart_id, exp_id, exp_data }: { chart_id: string
             })),
     };
 
+    const legendOnClickHandler = function (event:any, legendItem:any, legend:any) {
+        let index = legendItem.datasetIndex;
+        let ci = legend.chart;
+        let meta = ci.getDatasetMeta(index);
+
+        // Get the indexes of datasets with hidden === false
+        let visibleIndexes = ci.data.datasets.reduce(function(acc:any, dataset:any, i:any) {
+            let otherMeta = ci.getDatasetMeta(i);
+            if (otherMeta.hidden === false) {
+                acc.push(i);
+            }
+            return acc;
+        }, []);
+
+        // Toggle visibility of clicked dataset
+        if(meta.hidden === false) {
+            if(visibleIndexes.length === 1) {
+                ci.data.datasets.forEach(function(dataset:any, i:any) {
+                    let otherMeta = ci.getDatasetMeta(i);
+                    otherMeta.hidden = null;
+                });
+            }
+            else {
+                if(meta.hidden === false) {
+                    meta.hidden = true;
+                }
+            }
+        }
+        else {
+            meta.hidden = false;
+            ci.data.datasets.forEach(function(dataset:any, i:any) {
+                let otherMeta = ci.getDatasetMeta(i);
+                if (otherMeta.hidden === null && i !== index) {
+                    otherMeta.hidden = true;
+                }
+            });
+        }
```

Comment:
adding maybe a bit  more comments here to explain how exactly it "Toggle visibility of clicked dataset" ?

Reply:
Added!